src/datasources/system/SystemDataSource.ts

```diff
@@ -40,6 +40,13 @@ export class SystemDataSource extends DataSourceApi<SystemQuery> {
     return NetworkUtils.getIpAddressFromInterfaces(ip4Interface) || NetworkUtils.getIpAddressFromInterfaces(ip6Interface);
   }
 
+  async metricFindQuery(query: VariableQuery, options?: any) {
```

Comment:
This function needs a return type annotation

Reply:
Also, most of the example usage I found of `metricFindQuery` returned `Promise<MetricFindValue>`, where this currently returns simple `MetricFindValue`. Are you sure this works as intended?